streaming/base/stream.py

```diff
@@ -98,8 +99,11 @@ def __init__(self,
                  keep_zip: Optional[bool] = None) -> None:
         self.remote = remote
         self._local = local
-        self.local = local or mkdtemp()
         self.split = split or ''
+        if local is None:
```

Comment:
What would happen if user instantiate two or more `StreamingDataset` with same `remote` and `local=None` ? Ideal behavior is to throw a valid error message asking user to provide a `local` directory.

Reply:
With multiple `StreamingDataset` and multiple GPUs, how do we know if a directory is being created/accessed by the different GPUs within the same `StreamingDataset` or by different `StreamingDataset`s? Does streaming have functions similar to composer's [broadcast](https://github.com/mosaicml/composer/blob/dev/composer/utils/dist.py#L248C8-L248C8)? Should we use torch.dist?